op-conductor/rpc/ws/flashblocks_handler.go

```diff
@@ -0,0 +1,360 @@
+	clientsMu sync.RWMutex
+
+	// Register requests from the clients
+	register chan *Client
+
+	// Unregister requests from clients
+	unregister chan *Client
+
+	// Inbound messages to broadcast to the clients
+	broadcast chan []byte
+
+	// Signal to stop the hub
+	done chan struct{}
+}
+
+// newHub creates a new hub
+func newHub() *Hub {
+	return &Hub{
+		broadcast:  make(chan []byte, 256),
+		register:   make(chan *Client),
+		unregister: make(chan *Client, 64), // Add buffer to handle multiple unregister events
+		clients:    make(map[*Client]bool),
+		done:       make(chan struct{}),
+	}
+}
+
+// run starts the hub's main loop
+func (h *Hub) run() {
+	for {
+		select {
+		case <-h.done:
+			// Close all client connections
+			h.clientsMu.Lock()
+			for client := range h.clients {
+				client.Close()
+				delete(h.clients, client)
+			}
+			h.clientsMu.Unlock()
+			return
+		case client := <-h.register:
+			h.clientsMu.Lock()
+			h.clients[client] = true
+			h.clientsMu.Unlock()
+		case client := <-h.unregister:
+			h.clientsMu.Lock()
+			if _, ok := h.clients[client]; ok {
+				delete(h.clients, client)
+				client.Close()
+			}
+			h.clientsMu.Unlock()
+		case message := <-h.broadcast:
+			h.clientsMu.RLock()
+			for client := range h.clients {
+				select {
+				case client.send <- message:
+					// Message sent successfully
+					client.failureCount = 0 // Reset failure count on success
+				default:
+					// Client send buffer is full, increment failure count
+					client.failureCount++
+					if client.failureCount >= client.maxFailures {
+						// Send to unregister channel instead of directly manipulating the clients map
+						go func(c *Client) {
+							h.unregister <- c
+						}(client)
+					}
+				}
+			}
+			h.clientsMu.RUnlock()
+		}
+	}
+}
+
+// Handler implements the FlashblockHandler interface
+type Handler struct {
+	cfg             Config
+	log             log.Logger
+	isLeaderFn      func(context.Context) bool
+	rollupBoostConn *websocket.Conn
+	server          *http.Server
+	hub             *Hub
+}
+
+// NewHandler creates a new flashblocks handler
+func NewHandler(cfg Config, log log.Logger, isLeaderFn func(context.Context) bool) (FlashblockHandler, error) {
+	// Create the hub
+	hub := newHub()
+
+	// Initialize the handler
+	handler := &Handler{
+		cfg:        cfg,
+		log:        log,
+		isLeaderFn: isLeaderFn,
+		hub:        hub,
+	}
+
+	// Establish connection to rollup boost if URL is configured
+	if cfg.RollupBoostWsURL != "" {
+		log.Info("connecting to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+		conn, _, err := websocket.DefaultDialer.Dial(cfg.RollupBoostWsURL, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
+		}
+		handler.rollupBoostConn = conn
+		log.Info("connected to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
+	} else {
+		log.Info("rollup boost WebSocket disabled, no URL configured")
+	}
```

Comment:
```suggestion
	// Establish connection to rollup boost if URL is configured
	if cfg.RollupBoostWsURL == "" {
		log.Info("rollup boost WebSocket disabled, no URL configured")
		return
	} 
	
	log.Info("connecting to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
    conn, _, err := websocket.DefaultDialer.Dial(cfg.RollupBoostWsURL, nil)
    if err != nil {
	    return nil, fmt.Errorf("failed to connect to rollup boost WebSocket: %w", err)
    }
    handler.rollupBoostConn = conn
    log.Info("connected to rollup boost WebSocket", "url", cfg.RollupBoostWsURL)
```

nit, prefer this style to avoid if/else (potentially later nested if/elses)